modules/reconciliationRtdProvider.js

```diff
@@ -0,0 +1,313 @@
+    return null;
+  }
+}
+
+/**
+ * get all slots on page
+ * @return {Object[]} slot GoogleTag slots
+ */
+function getAllSlots() {
+  return utils.isGptPubadsDefined() && window.googletag.pubads().getSlots();
+}
+
+/**
+ * get GPT slot by placement id
+ * @param {string} code placement id
+ * @return {?Object}
+ */
+function getSlotByCode(code) {
+  const slots = getAllSlots();
+  if (!slots || !slots.length) {
+    return null;
+  }
+  return (
+    find(
+      slots,
+      (s) => s.getSlotElementId() === code || s.getAdUnitPath() === code
+    ) || null
+  );
+}
+
+/**
+ * get GPT slot by iframe window
+ * @param {Window} win
+ * @return {?Object}
+ */
+function getSlotByWin(win) {
+  const slots = getAllSlots();
+
+  if (!slots || !slots.length) {
+    return null;
+  }
+
+  return (
+    find(slots, (s) => {
+      let slotElement = document.getElementById(s.getSlotElementId());
+
+      if (slotElement) {
+        let slotIframe = slotElement.querySelector('iframe');
+
+        if (slotIframe && slotIframe.contentWindow === win) {
+          return true;
+        }
+      }
+
+      return false;
+    }) || null
+  );
+}
+
+/**
+ * serialize object and return query params string
+ * @param {Object} data
+ * @return {string}
+ */
+export function stringify(query) {
+  const parts = [];
+
+  for (let key in query) {
+    if (query.hasOwnProperty(key)) {
+      let val = query[key];
+      if (typeof query[key] !== 'object') {
+        parts.push(`${key}=${encodeURIComponent(val)}`);
+      } else {
+        parts.push(`${key}=${encodeURIComponent(stringify(val))}`);
+      }
+    }
+  }
+  return parts.join('&');
+}
+/**
+ * Init Reconciliation post messages listeners to handle
+ * impressions messages from ad creative
+ */
+function initListeners() {
+  window.addEventListener('message', handleAdMessage, false);
+}
+
+/**
+ * Send init event to log
+ * @param {Object} adUnitsDict
+ */
+function trackInit(adUnitsDict) {
+  const adUnits = Object.keys(adUnitsDict).map((k) => {
+    const adUnit = adUnitsDict[k];
+
+    return {
+      adUnitId: adUnit['RSDK_AUID'],
+      adDeliveryId: adUnit['RSDK_ADID'],
+    };
+  });
+
+  track.trackPost(
+    _moduleParams.initUrl,
+    {
+      adUnits,
+      publisherDomain: window.location.hostname,
+      publisherMemberId: _moduleParams.publisherMemberId,
+    }
+  );
+}
+
+/**
+ * Track event via POST request
+ * wrap method to allow stubbing in tests
+ * @param {string} url
+ * @param {Object} data
+ */
+export const track = {
+  trackPost(url, data) {
+    const ajax = ajaxBuilder();
+
+    ajax(
+      url,
+      function() {},
+      JSON.stringify(data),
+      {
+        method: 'POST',
+      }
+    );
+  }
+}
+
+/**
+ * Set custom targetings for provided adUnits
+ * @param {string[]} adUnitsCodes
+ * @return {Object} key-value object with custom targetings
+ */
+function getReconciliationData(adUnitsCodes) {
+  let dataToReturn = adUnitsCodes.reduce((rp, adUnitCode) => {
```

Comment:
I'd suggest changing this to `adUnitsCodes.forEach` and within that loop, populate 2 different structures:
```
dataToReturn - object
adUnitsToTrack - list of objects
```
`adUnitsToTrack` would be the structure needed in the `trackInit` function. This would avoid using reduce to create 1 structure and then map to create the 2nd, both from the same data.

Reply:
@msm0504, thank you.
Modified the loop.